### ISSUE TITLE
auto_accounts needs a relationship to auto_queues

### ISSUE THREAD

**Issue by johrstrom** [OSC/ondemand#3027]
From discourse: https://discourse.openondemand.org/t/struggling-to-enable-dynamic-forms/2946

It seems every dynamic account should also populate `data-option-for-auto-queue-<QUEUE_NAME>`.

**Comment by Micket** [comment]
Yes, something along this line, the entire "cluster:partition:account:qos" relationship has potentially only partially valid options. I don't happen to use qos this way, but, i think it's possible. 

To represent this with data-option-for-xxx one would have to pick a consistent order (doesn't really matter of account has limitations on account or the other way around), but to avoid deadlocks I think one needs to ensure the strategy is something like:
1. cluster (may not have any conditions)
2. partition (may be conditional on cluster)
3. qos (may be conditional on cluster and partition)
4. account (may be conditional on cluster, partition, qos)

otherwise one risks selecting a qos that doens't allow and account, that doesn't allow a paritition that doesn't allow any qos, or some strange deadlock like that where nothing is selectable.

I'll just note that with a setup with private partitions, most users will only ever have access to at most 2 (main partition and maybe 1 private partition) out of ~10 partitions, so best would be not to present the options that don't have any valid accounts at all. 
I guess this could apply to cluster, partitions and qos all the same.

**Comment by johrstrom** [comment]
I'm looking into this a bit now. I think I'll have to revert #2594 and allow for duplicates in any given list to be populated.

**Comment by johrstrom** [comment]
I put some work into this for the 3.1 release, though I can't say it's finished. If you could provide some sacctmgr output or sinfo output or similar to test against that'd be super helpful. We don't really expose queues to our customers so it's I'm struggling a bit to meet the needs of sites that do without something to test against.

**Comment by johrstrom** [comment]
Yea we're still kinda looking for sacctmgr output or sinfo output or similar to test against.  We don't use queues like many other centers do, so it's very hard to develop features like this given we don't have anything near what's expected.

**Comment by moffatsadeghi** [comment]
@simonLeary42 does this issue tie into your PRs in ood_core and ondemand? 

We are reviewing final items to complete for 4.1.  Do you have an anticipated timeline for various PRs? :)

**Comment by moffatsadeghi** [comment]
Review for 4.1 notes: There is an importance to complete this issue so that we can step up our support for centers.  We do want to this to be completed for 4.1, but depending on the timeline, this may get bumped, unfortunately.

**Comment by simonLeary42** [comment]
> We are reviewing final items to complete for 4.1. Do you have an anticipated timeline for various PRs? :)

@johrstrom said in Slack that he would take them over the finish line, as it were: https://openondemand.slack.com/archives/CEYDKTWMT/p1755875812724939

**Comment by simonLeary42** [comment]
I pretty much offloaded all this stuff from my head so I'm cleaning out the cobwebs...

https://github.com/OSC/ondemand/pull/4471 does tie in here, but not in a very big way. Each option in `auto_queues` already contains `data-option-for-auto-accounts-...` based on OOD's understanding of account/queue access info. Since Slurm queue access can also be done on a QoS basis, and Slurm QoS are granted to accounts, I made the Slurm API aware of this, and updated `account_cache.account_allowed?` so that if a queue requires certain QoSes, and this account does not have access to any of those QoSes, then this account does not have access to that queue.

I have 6 open PRs that deal with the Slurm API, which would have to be merged in a certain order:

```mermaid
graph LR;
classDef red stroke:#f00
classDef green stroke:#0f0
classDef blue stroke:#00f
classDef orange stroke:#fb0
ood_core#892:::orange
ood_core#888:::red
ondemand#4471:::red
ood_core#890:::blue
ondemand#4475:::blue
%% ood_core#895/904:::green
%% ondemand#4491:::green
ondemand#4698:::green

ood_core#888 --> ondemand#4471
ood_core#890 --> ondemand#4475
%% ood_core#890 --> ood_core#895/904
%% ondemand#4475 --> ondemand#4491
```

* orange: removing obscure unused data from slurm API
* red: described above
* blue: remove misleading `AccountInfo.queue`
* green: fix assumption that the 1st cluster in the list contians the accounts for all clusters

**Comment by simonLeary42** [comment]
Linking:
* orange
    * https://github.com/OSC/ood_core/pull/892
* red
    * https://github.com/OSC/ood_core/pull/888
    * https://github.com/OSC/ondemand/pull/4471
* blue
    * https://github.com/OSC/ood_core/pull/890
    * https://github.com/OSC/ondemand/pull/4475
* green
    * ~~https://github.com/OSC/ood_core/pull/895~~ https://github.com/OSC/ood_core/pull/904
    * https://github.com/OSC/ondemand/pull/4491
    * https://github.com/OSC/ondemand/pull/4698
